src/Controls/src/Core/CompatServiceProvider.cs

```diff
@@ -0,0 +1,55 @@
+#nullable enable
+using System;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Maui.Hosting.Internal;
+
+namespace Microsoft.Maui.Controls
+{
+	// TODO: Remove this and use the real service provider.
+	static class CompatServiceProvider
```

Comment:
This "service provider" is just a way for us to access the Core services if they were set, but if this is a pure compat thing and no migration was done, we get the defaults. This allows for simple usage where we assume this is available. This is sort of the pseudo `Forms` and `Registrar` types where things were chucked on. No new things should be added here unless they are being migrated out of some static thing. This might be other dependency service registration from the compat world.

Reply:
I don't think "a pure compat thing" is a supported scenario

Like whatever they migrate over to will still be a full MAUI app and then compat will only be there if they want to register compat renderers

Reply:
Maybe. This was more for people that switch over to Maui and not have the app crash out the gate if they don't have a startup and all the builder things. This is still used by the compat renderers to access things like the font manager and services since they are still a static world. But this should go away once we get those to use the builder.